contracts/strategies/QVSimple.sol

```diff
@@ -0,0 +1,270 @@
+        if (!distributionStarted) {
+            distributionStarted = true;
+        }
+    }
+
+    /// @notice Allocate voice credits to an array of recipients
+    /// @param __recipients The recipients
+    /// @param _amounts The amounts of voice credits to allocate
+    /// @param _data The data
+    /// @param _sender The actual sender of the transaction
+    function _allocate(address[] memory __recipients, uint256[] memory _amounts, bytes memory _data, address _sender)
+        internal
+        override
+    {
+        uint256 voiceCreditsAlreadyAllocated = voiceCreditsAllocated[_sender];
+
+        // check that the sender can allocate votes
+        if (!_isValidAllocator(_sender)) revert UNAUTHORIZED();
+
+        uint256 voiceCreditsToAllocate;
+
+        for (uint256 i = 0; i < __recipients.length; i++) {
+            // check the voice credits to allocate is > 0
+            if (_amounts[i] == 0) revert INVALID();
+
+            // check that the recipient is accepted
+            if (!_isAcceptedRecipient(__recipients[i])) revert RECIPIENT_ERROR(__recipients[i]);
+
+            // sum up the voice credits to allocate
+            voiceCreditsToAllocate += _amounts[i];
+
+            emit Allocated(__recipients[i], _sender, voiceCreditsToAllocate, _data);
+        }
+
+        // check that the allocator has voice credits left to allocate
+        if (!_hasVoiceCreditsLeft(voiceCreditsToAllocate, voiceCreditsAlreadyAllocated)) revert INVALID();
+
+        _votingState._voteWithVoiceCredits(__recipients, _amounts);
+
+        voiceCreditsAllocated[_sender] += voiceCreditsToAllocate;
+    }
+
+    /// @notice Returns if the recipient is accepted
+    /// @param _recipientId The recipient id
+    /// @return true if the recipient is accepted
+    function _isAcceptedRecipient(address _recipientId) internal view returns (bool) {
+        return _getRecipientStatus(_recipientId) == Status.Accepted;
+    }
+
+    /// @notice Checks if the allocator is valid
+    /// @param _allocator The allocator address
+    /// @return true if the allocator is valid
+    function _isValidAllocator(address _allocator) internal view returns (bool) {
+        return allowedAllocators[_allocator];
+    }
+
+    /// @notice Checks if the allocator has voice credits left
+    /// @param _voiceCreditsToAllocate The voice credits to allocate
+    /// @param _allocatedVoiceCredits The allocated voice credits
+    /// @return true if the allocator has voice credits left
+    function _hasVoiceCreditsLeft(uint256 _voiceCreditsToAllocate, uint256 _allocatedVoiceCredits)
+        internal
+        view
+        returns (bool)
+    {
+        return _voiceCreditsToAllocate + _allocatedVoiceCredits <= maxVoiceCreditsPerAllocator;
+    }
+
+    /// @notice Ensure no withdrawals are allowed before the allocation end time
+    function _beforeWithdraw(address, uint256, address) internal override {
+        if (block.timestamp <= allocationEndTime + 30 days) {
```

Comment:
maybe instead of having +30 days we can add this lock window as an initialized param and define it on initialization?